### PR TITLE
treewide/nixos: remove `with lib;` part 5

### PR DESCRIPTION
**Pull request by Stunkymonkey**
## Description of changes

part of #208242

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by philiptaron** [comment]
Felix, is this with the new script or old?

**Comment by Stunkymonkey** [comment]
@philiptaron this was done with the new one. But the commit messages are not yet improved...

**Comment by philiptaron** [comment]
If we split this part 5 and the part 6 into groups of 50, I think I can reviews them all this coming week.

**Comment by Aleksanaa** [comment]
Do we have an AST analyzer (with eval?) now, or still manually?

File: nixos/modules/services/databases/cockroachdb.nix [hunk 1]
```diff
@@ -1,12 +1,9 @@
 { config, lib, pkgs, utils, ... }:
-
-with lib;
-
 let
   cfg = config.services.cockroachdb;
   crdb = cfg.package;
 
-  startupCommand = utils.escapeSystemdExecArgs
+  startupCommand = utils.lib.escapeSystemdExecArgs
```
Comment (philiptaron) [inline nixos/modules/services/databases/cockroachdb.nix:6]: revert

File: nixos/modules/services/databases/redis.nix [hunk 2]
```diff
@@ -131,74 +128,74 @@ in {
...
             };
 
-            save = mkOption {
-              type = with types; listOf (listOf int);
+            save = lib.mkOption {
+              type = with lib.types; listOf (listOf int);
               default = [ [900 1] [300 10] [60 10000] ];
               description = ''
-                The schedule in which data is persisted to disk, represented as a list of lists where the first element represent the amount of seconds and the second the number of changes.
+                The schedule in which data is persisted to disk, represented as a list of lib.lists where the first lib.element represent the amount of seconds and the second the number of changes.
```
Comment (philiptaron) [inline nixos/modules/services/databases/redis.nix:183]: revert

File: nixos/modules/services/databases/redis.nix [hunk 3]
```diff
@@ -131,74 +128,74 @@ in {
...
 
-                If set to the empty list (`[]`) then RDB persistence will be disabled (useful if you are using AOF or don't want any persistence).
+                If set to the empty list (`[]`) then RDB persistence will be disabled (useful if you are using AOF or don't want lib.any persistence).
```
Comment (philiptaron) [inline nixos/modules/services/databases/redis.nix:185]: revert

File: nixos/modules/services/desktops/geoclue2.nix [hunk 4]
```diff
@@ -80,95 +76,84 @@ in
...
+      geoProviderUrl = lib.mkOption {
+        type = lib.types.str;
         default = "https://location.services.mozilla.com/v1/geolocate?key=geoclue";
         example = "https://www.googleapis.com/geolocation/v1/geolocate?key=YOUR_KEY";
         description = ''
           The url to the wifi GeoLocation Service.
         '';
       };
 
-      package = mkOption {
```
Comment (philiptaron) [inline nixos/modules/services/desktops/geoclue2.nix:132]: bad merge

**Review by philiptaron — CHANGES_REQUESTED** [review]
Felix, there's a lot of defects in this tranche. What's different here from part 3, which was defect free?

File: nixos/modules/installer/sd-card/sd-image.nix [hunk 5]
```diff
@@ -10,53 +10,49 @@
...
-  } // optionalAttrs (config.sdImage.rootPartitionUUID != null) {
+  } // lib.optionalAttrs (config.sdImage.rootPartitionUUID != null) {
     uuid = config.sdImage.rootPartitionUUID;
   });
 in
 {
   imports = [
-    (mkRemovedOptionModule [ "sdImage" "bootPartitionID" ] "The FAT partition for SD image now only holds the Raspberry Pi firmware files. Use firmwarePartitionID to configure that partition's ID.")
-    (mkRemovedOptionModule [ "sdImage" "bootSize" ] "The boot files for SD image have been moved to the main ext4 partition. The FAT partition now only holds the Raspberry Pi firmware files. Changing its size may not be required.")
+    (lib.mkRemovedOptionModule [ "sdImage" "bootPartitionID" ] "The FAT lib.partition for SD image now only holds the Raspberry Pi firmware files. Use firmwarePartitionID to configure that lib.partition's ID.")
```
[the remark this answers is not in the document]
Comment (Stunkymonkey) [reply nixos/modules/installer/sd-card/sd-image.nix:26]: hmmm looks to me like all the old changes are still in. the new script could not have produced this.

[the remark this answers is not in the document]
Comment (Stunkymonkey) [reply nixos/modules/installer/sd-card/sd-image.nix]: @philiptaron i rerun the script.

[the remark this answers is not in the document]
Comment (Stunkymonkey) [reply nixos/modules/installer/sd-card/sd-image.nix]: had to remove some commits (only removing lines), because they were already fixed upstream.

**Review by philiptaron — APPROVED** [review]
Whew, it's at zero defects this time.

**Comment by philiptaron** [comment]
> @philiptaron can we agree, that when you find a single defect in this kind of MRs -> mention it an abort any further reading?

Sure, I can do that. Now that we have a script that _ought_ to reliably yield zero defects that's a sane default.

**Comment by philiptaron** [comment]
No error reports the next day. Looking at tranche 6 now.

**Comment by K900** [comment]
https://github.com/NixOS/nixpkgs/pull/363775

How many of these do we need to explode to realize that we need better testing? I believe we're currently at 4/5.

**Comment by philiptaron** [comment]
@K900, the PRs done with the old sed-based script were definitely too low quality. We paused any merges of changes produced with that tool.

So please do not count those initial failures as being part of this one.

The new script that Felix uses in this PR is subject to time-of-script-run issues, where unless there is a merge conflict (not guaranteed) things like https://github.com/NixOS/nixpkgs/pull/363775 are possible.

On a personal note, I find the characterization of one quickly fixed defect across ~1,000 LOC change as an "explosion" to be unkind. You yourself have made many defects, merged them to master, then fixed them. That's good. Please don't harangue others for behavior you engage in.

I'm all ears for a testing strategy that produces fewer defects, that is actually able to be run by humans without a build lab. It's certainly not my intention to check in code that causes defects.

**Comment by Mic92** [comment]
@philiptaron I think you can run this:

```
nix-eval-jobs --workers <replaceme> --force-recurse --flake .#legacyPackages.x86_64-linux.nixosTests --force-recurse
```

Replace `replaceme` with: maximum_ram_usage / 4GB

**Comment by Stunkymonkey** [comment]
i can confirm my script completely ignores the line in #363768.

We should have run `nix-build . -A nixosTests.redis` to prevent this. Are there any other commands that are more "leightweight" to detect this? i tried `nix-instantiate --parse` but no success.

**Comment by Mic92** [comment]
Not sure if light-weight alternatives give you decent coverage. Our nix-community x86 machine might be a good target to run this for a bit longer.

**Pull request merged**